test/e2e/cross_os/test_job_submissions.py

```diff
@@ -1437,3 +1437,160 @@ def check_percentage(complete_percentage) -> bool:
+
+        # Submit a job with a task that sleeps for 60 seconds , which is more than the UpdateWorkerSchedule interval of 30 seconds
+
+        test_run_status_message = "Sleep job is running!"
+        sleep_script = (
+            f"""
+            #!/usr/bin/env bash
+            percent=0
+            
+            while [ $percent -le 100 ] 
+            do
+                echo "openjd_progress: $percent"
+                echo "openjd_status: {test_run_status_message}"
+                ((percent+=10))
+                sleep 6
+            done
+            """
+            if os.environ["OPERATING_SYSTEM"] == "linux"
+            else f"""
+            $percent = 0
+            while ($percent -le 100) {{
+                Write-Output "openjd_progress: $percent"
+                Write-Output "openjd_status: {test_run_status_message}"
+                $percent += 10
+                Start-Sleep -Seconds 6
+            }}
+            """
+        )
+        job: Job = Job.submit(
+            client=deadline_client,
+            farm=deadline_resources.farm,
+            queue=deadline_resources.queue_a,
+            priority=98,
+            template={
+                "specificationVersion": "jobtemplate-2023-09",
+                "name": "One Minute Sleep Job for Task Progress",
+                "steps": [
+                    {
+                        "name": "60 Second Sleep",
+                        "hostRequirements": {
+                            "attributes": [
+                                {
+                                    "name": "attr.worker.os.family",
+                                    "allOf": [os.environ["OPERATING_SYSTEM"]],
+                                }
+                            ]
+                        },
+                        "script": {
+                            "actions": {
+                                "onRun": (
+                                    {"command": "{{ Task.File.runScript }}"}
+                                    if os.environ["OPERATING_SYSTEM"] == "linux"
+                                    else {
+                                        "command": "powershell",
+                                        "args": ["{{ Task.File.runScript }}"],
+                                    }
+                                ),
+                            },
+                            "embeddedFiles": [
+                                {
+                                    "name": "runScript",
+                                    "type": "TEXT",
+                                    "runnable": True,
+                                    "data": sleep_script,
+                                    **(
+                                        {"filename": "SleepScript.ps1"}
+                                        if os.environ["OPERATING_SYSTEM"] == "windows"
+                                        else {}
```

Comment:
Shouldn't this be a filename like `SleepScript.sh` or something as well?

Reply:
Only windows need a file to run. Linux does not

Reply:
This filename is an optional parameter. We use it here on Windows so the script runs as powershell and not a batch script.

https://github.com/OpenJobDescription/openjd-specifications/wiki/2023-09-Template-Schemas#61-embeddedfiletext

> filename — The filename for the written file. This must strictly be the basename of the filename, and not contain any directory pathing. (i.e. foo.txt not dir/foo.txt). Defaults to a random filename if not provided.